cpp/src/tiledb/cxx_api/tiledb_cxx_array_util.cc

```diff
@@ -9,33 +9,49 @@ std::string ArrayUtil::get_tiledb_version()
   return std::to_string(t[0]) + "." + std::to_string(t[1]) + "." + std::to_string(t[2]);
 }//std::string ArrayUtil::get_tiledb_version()
 
-int ArrayUtil::export_file_to_path(const std::string& file_uri, const std::string& output_path, const std::shared_ptr<tiledb::Context>& ctx)
+int ArrayUtil::export_file_to_path(const std::string& file_uri, const std::string& output_path, uint64_t buffer_size, const std::shared_ptr<tiledb::Context>& ctx)
 {
   const std::string METADATA_SIZE_KEY = "file_size";
   const std::string FILE_ATTRIBUTE_NAME = "contents";
+  const std::string METADATA_ORIGINAL_FILE_NAME = "original_file_name";
   try {
-    std::shared_ptr<tiledb::VFS> vfs = std::shared_ptr<tiledb::VFS>(new tiledb::VFS(ctx));
-    tiledb::impl::VFSFilebuf filebuf = tiledb::impl::VFSFilebuf(vfs);
-    filebuf.open(output_path, std::ios::out);
-    std::ostream os(&filebuf);
+
 
     std::shared_ptr<Array> array = std::shared_ptr<Array>(new Array(ctx, file_uri, tiledb::QueryType::TILEDB_READ));
 
+    
     //get file_size from metadata
     uint32_t v_num;
     const void* v_r;
     tiledb::DataType file_size_datatype = tiledb::DataType::TILEDB_UINT64;
     array->get_metadata(METADATA_SIZE_KEY, &file_size_datatype, &v_num, &v_r);
     uint64_t file_size = *((const int64_t*)v_r);
+    if(buffer_size==0) {
+      buffer_size = file_size;
+    }
+
+    std::string output_file = output_path;
+    if(output_file.empty()) {
+      tiledb::DataType file_name_datatype = tiledb::DataType::TILEDB_STRING_ASCII;
+      uint32_t name_size = 0;
+      char original_name[1024] = "";
+      array->get_metadata(METADATA_ORIGINAL_FILE_NAME,&file_name_datatype,&name_size,reinterpret_cast<const void**>(&original_name));
+      output_file = std::string(original_name);
```

Comment:
This should be `output_file = std::string(original_name, name_size)`

Reply:
corrected.